store/displayPreferences.ts

```diff
@@ -0,0 +1,224 @@
+   */
+  INIT_STATE(
+    state: DisplayPreferencesState,
+    { displayPreferences }: { displayPreferences: DisplayPreferencesDto }
+  ) {
+    Object.assign(state, displayPreferences);
+    for (const key in defaultCustomPrefs) {
+      if (!(key in state.CustomPrefs)) {
+        state.CustomPrefs[key] = defaultCustomPrefs[key];
+      }
+    }
+  },
+
+  /**
+   * Edits a custom pref key
+   *
+   * @param {DisplayPreferencesState} state Current state
+   * @param {any} param1 Payload
+   * @param {SingleCustomPrefMutationPayload} param1.pref Key and value for new custom pref
+   */
+  EDIT_CUSTOM_PREF(
+    state: DisplayPreferencesState,
+    { pref }: { pref: SingleCustomPrefMutationPayload }
+  ) {
+    state.CustomPrefs[pref.key] = pref.value;
+    if (pref.key in updateMethods) updateMethods[pref.key](pref.value);
+  },
+
+  CALL_ALL_CALLBACKS(state: DisplayPreferencesState) {
+    Object.keys(state.CustomPrefs).forEach((key) => {
+      if (key in updateMethods) updateMethods[key](state.CustomPrefs[key]);
+    });
+  }
+};
+
+export const actions: ActionTree<
+  DisplayPreferencesState,
+  DisplayPreferencesState
+> = {
+  /**
+   * Fetches display preferences, stores them and updates Vuetify dark mode
+   *
+   * @param {any} param0 Vuex
+   * @param {any} param0.commit Vuex commit
+   * @param {any} param0.dispatch Vuex dispatch
+   */
+  async initState({ commit, dispatch }) {
+    try {
+      const response = await this.$api.displayPreferences.getDisplayPreferences(
+        {
+          displayPreferencesId: 'usersettings',
+          userId: this.$auth.user.Id,
+          client: 'vue'
+        }
+      );
+
+      if (response.status !== 200)
+        throw new Error(
+          'get display preferences status response = ' + response.status
+        );
+
+      commit('INIT_STATE', { displayPreferences: response.data });
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      const message = this.$i18n.t('failedRetrievingDisplayPreferences');
+      dispatch(
+        'snackbar/pushSnackbarMessage',
+        { message, color: 'error' },
+        { root: true }
+      );
+    }
+    commit('CALL_ALL_CALLBACKS');
+  },
+
+  /**
+   * Pushes the current state to the server
+   *
+   * @param {any} param0 Vuex
+   * @param {any} param0.state Vuex state
+   * @param {any} param0.dispatch Vuex dispatch
+   */
+  async pushState({ state, dispatch }) {
+    try {
+      const response = await this.$api.displayPreferences.updateDisplayPreferences(
+        {
+          displayPreferencesId: 'usersettings',
+          userId: this.$auth.user.Id,
+          client: 'vue',
+          displayPreferencesDto: state
+        }
+      );
+      if (response.status !== 204)
+        throw new Error(
+          'set display preferences status response = ' + response.status
+        );
+    } catch (error) {
+      // eslint-disable-next-line no-console
```

Comment:
The `initState` has the same error handling as this one. We can extract this in action and just trigger it in both places.

Reply:
Done